dev/source/docs/ros2-over-ethernet.rst

```diff
@@ -0,0 +1,93 @@
+.. _ros2-over-ethernet:
+
+===================
+ROS 2 over Ethernet
+===================
+
+The following tutorial explains how to connect a flight controller to a companion computer using ethernet for DDS communication.
+
+Ensure you have ROS 2 :ref:`installed <ros2>` and have run :ref:`SITL <ros2-sitl>` successfully before attempting this page.
+
+Additionally, make sure you understand the :ref:`basics of networking in ArduPilot. <common-network>`
+
+==========
+Motivation
+==========
+
+Starting in ArduPilot 4.5, ethernet is now supported as a new communication interface.
+Compared to connecting an autopilot over serial, ethernet has the following advantages:
+
+* Ethernet is more immune to noise because it uses twisted pair wiring
+* Ethernet can run multiple protocols over the same protocol (unless you use PPP)
+* Ethernet is easier to debug with tools such as ``tcpdump`` or ``Wireshark``
+* Ethernet uses standard cabling, so it's more difficult to mix up TX and RX
+
+===============
+Physical Equipment
+===============
+
+The following equipment is required to complete this tutorial:
+
+* A computer that can run the MicroROS Agent and the ROS 2 CLI
+* An autopilot with ethernet support
```

Comment:
Is DDS compiled into ArduPilot by default? I'd be kind of surprised. At very least it must require a 2M board no?

Reply:
No, it's mentioned in the pre-reqs to use `--enable-dds` when compiling.

Reply:
I added a reminder in the steps list to use `--enable-dds`

Reply:
Not seeing it. Where is that?

Reply:
Commit 6c70a032d98b83650c4ddde600115e3b24236a81 just now.